IdentityCore/src/oauth2/aad_v1/MSIDAADV1Oauth2Factory.m

```diff
@@ -151,15 +152,70 @@ - (MSIDLegacySingleResourceToken *)legacyTokenFromResponse:(MSIDTokenResponse *)
 #pragma mark - Webview controllers
 - (id<MSIDWebviewInteracting>)embeddedWebviewControllerWithRequest:(MSIDRequestParameters *)requestParams
                                                            Webview:(WKWebView *)webview
+                                                 completionHandler:(MSIDWebUICompletionHandler)completionHandler
 {
     // Create MSIDEmbeddedWebviewRequest and create EmbeddedWebviewController
     return nil;
 }
 
 - (id<MSIDWebviewInteracting>)systemWebviewControllerWithRequest:(MSIDRequestParameters *)requestParams
+                                               callbackURLScheme:(NSString *)callbackURLScheme
+                                               completionHandler:(MSIDWebUICompletionHandler)completionHandler
 {
     // Create MSIDSystemWebviewRequest and create SystemWebviewController
     return nil;
 }
 
+- (NSURL *)startURLFrom:(MSIDRequestParameters *)requestParams
+{
+    NSString *state = [self encodeProtocolState:requestParams];
+    
+    // if value is nil, it won't appear in the dictionary
+    NSMutableDictionary *queryParams = [NSMutableDictionary dictionaryWithObjectsAndKeys:
+                                        MSID_OAUTH2_CODE, MSID_OAUTH2_RESPONSE_TYPE,
+                                        [requestParams clientId], MSID_OAUTH2_CLIENT_ID,
+                                        [requestParams resource], MSID_OAUTH2_RESOURCE,
+                                        [requestParams redirectUri], MSID_OAUTH2_REDIRECT_URI,
+                                        state, MSID_OAUTH2_STATE,
+                                        requestParams.promptBehavior, @"prompt",
+                                        @"1", @"hashchrome", //to hide back button in UI
```

Comment:
just curious, do we still need this?

Reply:
This code is JZ's code. I was merely copying and pasting for now to make it compile.
Will remove this code for now from mine.